### ISSUE TITLE
Fail to start when using an initializer with agent Yarn

### ISSUE THREAD

**Issue by mikelorant** [slidevjs/slidev#1233]
**Describe the bug**
Slidev fails to start when using an initializer with the agent set to Yarn.

**To Reproduce**
1. Run the command `npm init slidev@0.46.1`.
2. When prompted `Project name` respond with `slidev`.
3. When prompted `Install and start it now?` respond with `yes`.
4. When prompted `Choose the agent` select `yarn`.

The following error will be displayed when the link to the slides open in the web browser.
```
11:20:05 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js
11:20:06 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js (x2)
11:20:06 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js (x3)
11:20:08 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js (x4)
11:20:08 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js (x5)
11:20:08 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js (x6)
11:20:08 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js (x7)
11:20:09 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js (x8)
11:20:09 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js (x9)
11:20:09 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js (x10)
11:20:09 am [vite] Pre-transform error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js (x11)
11:20:09 am [vite] Internal server error: Cannot find module 'typescript'
Require stack:
- /slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js
  Plugin: unplugin-vue-markdown
  File: /@slidev/slides/5.md
      at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
      at Module._load (node:internal/modules/cjs/loader:984:27)
      at Module.require (node:internal/modules/cjs/loader:1234:19)
      at require (node:internal/modules/helpers:176:18)
      at twoslasher (/slidev/node_modules/@typescript/twoslash/dist/twoslash.cjs.development.js:606:81)
      at Object.preprocess (file:///slidev/node_modules/shikiji-twoslash/dist/core.mjs:782:28)
      at codeToHast (file:///slidev/node_modules/shikiji-core/dist/index.mjs:3917:41)
      at codeToHtml (file:///slidev/node_modules/shikiji-core/dist/index.mjs:4151:25)
      at Object.codeToHtml (file:///slidev/node_modules/shikiji-core/dist/index.mjs:4931:40)
      at markdownit.options.highlight (file:///slidev/node_modules/markdown-it-shikiji/dist/core.mjs:56:24)
      at default_rules.fence (/slidev/node_modules/markdown-it/lib/renderer.js:53:27)
      at Renderer.render (/slidev/node_modules/markdown-it/lib/renderer.js:332:28)
      at MarkdownIt.render (/slidev/node_modules/markdown-it/lib/index.js:544:24)
      at file:///slidev/node_modules/unplugin-vue-markdown/dist/chunk-EES4WMNF.js:242:25
      at TransformContext.transform (file:///slidev/node_modules/unplugin-vue-markdown/dist/chunk-EES4WMNF.js:413:37)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async Object.transform (file:///slidev/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:63705:30)
      at async loadAndTransform (file:///slidev/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:49386:29)
      at async viteTransformMiddleware (file:///slidev/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:58987:32)
```

Twoslash support was added to this version and the example on slide 5 was extended to showcase this feature.

**Variations**

| Version  | NPM     | Yarn    |
| -------- | ------- | ------- |
| `0.45.0` | Success | Success |
| `0.46.1` | Success | Fail    |

Attempting to use version `0.46.0` for the initializer wasn't useful as this still installed version `0.46.1` which matched the same results above. The version of initializer is not important, it is the release that is installed that seems to determine the outcome.

**Possible Cause**

It seems pull request #1218 is where this bug was added. It is clear from the pull request itself that tests were flaky and reporting inconsistent problems. Under such circumstance I do not believe this pull request should have been merged in this state.

When release `0.46.1` was tagged it is also shown that tests were failing on macOS.

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Firedox
 - Slidev version: 0.46.1
 - NodeJS version:  21.5.0
 - Yarn version: 1.22.21

**Comment by ayanamists** [comment]
I also experience the same problem, with issue #1223  reported.

**Comment by bayasdev** [comment]
Can confirm this problem doesn't happen when bootstrapping the project with `npm`

**Comment by KermanX** [comment]
As antfu said in https://github.com/slidevjs/slidev/issues/1223#issuecomment-1924190944, hopefully this should be fixed in 0.47. Close this for now.

**Comment by mikelorant** [comment]
Confirmed resolved.
